### PR TITLE
Feature/mc 12978 create workload

### PR DESCRIPTION
**Pull request by RGitB**
#### Changes made
- Added a section for creating a workload, with examples

#### Related PRs
- [20](https://github.com/cloudops/cloudmc-stackpath-plugin/pull/20)

### CODE REVIEW

File: source/includes/stackpath/_workloads.md [hunk 1]
```diff
@@ -117,6 +117,103 @@ Attributes | &nbsp;
...
+   "deploymentPops":"FRA",
+   "enableAutoScaling":true,
+   "cpuUtilization":50,
+   "minInstancesPerPop":1,
+   "maxInstancesPerPop":2
+}
+```
+<code>POST /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/workloads</code>
+
+Create a new workload
```
Comment (patrickspensieri) [inline source/includes/stackpath/_workloads.md]: Should be `Create a new workload in a given [environment](#administration-environments).`

File: source/includes/stackpath/_workloads.md [hunk 2]
```diff
@@ -117,6 +117,103 @@ Attributes | &nbsp;
...
 
+<!-------------------- CREATE A WORKLOAD -------------------->
+### Create a workload
+```shell
+curl -X POST \
+    -H "MC-Api-Key: your_api_key" \
+    -d "request_body" \
+    "https://cloudmc_endpoint/v1/services/stackpath/test-area/workloads"
+```
+> Request body example for a VM:
```
Comment (patrickspensieri) [inline source/includes/stackpath/_workloads.md]: Just a small formatting change to match [our docs template](https://raw.githubusercontent.com/cloudops/cloudmc-api-docs/cmc-dev/.github/PULL_REQUEST_TEMPLATE.md) 📜 

To be consistent with the rest of the docs, we could put both request examples in the same code block (with format js) and the response example in another.

> Request body examples:
```js
// create a vm
{
   name: my-vm,
   slug: my-vm-slug,
   ...
}

// create a container
{
   name: my-container,
   slug: my-container-slug,
   ...
}
```

> The above commands return a JSON structured like this:
```json
{
  "taskId": "ef70cafa-0544-4709-a66a-c68595ee105a",
  "taskStatus": "SUCCESS"
}

```

File: source/includes/stackpath/_workloads.md (hunk 1 above)
Comment (RGitB) [reply source/includes/stackpath/_workloads.md]: good catch. Thanks Pat

File: source/includes/stackpath/_workloads.md (hunk 2 above)
Comment (RGitB) [reply source/includes/stackpath/_workloads.md]: fixed in next commit

**Review by patrickspensieri — APPROVED** [review]
Thanks for the quick fix ✔️

**Review by razine-bensari — APPROVED** [review]
lgtm!

File: source/includes/stackpath/_workloads.md [hunk 3]
```diff
@@ -117,6 +117,110 @@ Attributes | &nbsp;
...
+ `vpc`<br/>*string* | The virtual private cloud option for now supports default vpc only.
+ `protocol`<br/>*string* | Supported protocols are: `TCP`, `UDP` and `TCP_UDP`.
+ `firstBootSshKey(s)`<br/>*string* | If creating a VM-based workload, SSH keys are required. Multiple SSH keys can be separated by commas.
+ `specs`<br/>*string* | Specification type for resources which are allocated to each instance in a workload. Supported specifications are `SP-1 (1 vCPU, 2 GB RAM)`,`SP-2 (2 vCPU, 4 GB RAM)`,`SP-3 (2 vCPU, 8GB RAM)`,`SP-4 (4 vCPU, 16 GB RAM)`,`SP-5 (8 vCPU, 32 GB RAM)`.
+ `deploymentName`<br/>*string* | The name of the deployment.
+ `deploymentPops`<br/>*string* | The point of presence of a deployment. In the format [A-Z][A-Z][A-Z].
+ `enableAutoScaling` <br/>*boolean* | Specify if you would like to enable autoscaling. If enabled, the following fields are required.
```
Comment (y0geshdev) [inline source/includes/stackpath/_workloads.md]:
```suggestion
 `enableAutoScaling` <br/>*boolean* | Specify if you would like to enable autoscaling. If enabled, then `cpuUtilization` , `minInstancesPerPop` and `maxInstancesPerPop` are required.
```

We might want to add more fields after `maxInstancesPerPop` and this way it won't confuse API consumers if those newly added fields after `enableAutoScaling` are required by `enableAutoScaling` or not.

File: source/includes/stackpath/_workloads.md [hunk 4]
```diff
@@ -117,6 +117,110 @@ Attributes | &nbsp;
...
+ `maxInstancesPerPop` <br/>*int* | The maximum number of instances per PoP.
+ 
+ Optional | &nbsp;
+ ------- | -----------
+ `addAnyCastIpAddress`<br/>*boolean* | Option to AnyCast IP Address.
+ `slug`<br/>*string* | A workload's programmatic name. Workload slugs are used to build its instances names. If not provided, defaults to workload's name.
+ `publicPorts`<br/>*string* | A single port, such as 80 or a port range, such as 1024-65535.
+ `persistenceStoragePath`<br/>*string* | The path in an instance to mount a volume.
+ `persistenceStorageSize`<br/>*int* | The size of the mounted volume (in GB).
+ `addImagePullCredentialsOption` <br/>*boolean* | When creating a Container based workload, use this option to pull a container image.
```
Comment (y0geshdev) [inline source/includes/stackpath/_workloads.md]:
```suggestion
 `addImagePullCredentialsOption` <br/>*boolean* | It is used to indicate if additional credentials to pull container image are provided or not.
```

**Review by y0geshdev — CHANGES_REQUESTED** [review]
Left 2 minor suggestions to improve wording a bit.
Overall, it looks good. Good work.

File: source/includes/stackpath/_workloads.md (hunk 4 above)
Comment (RGitB) [reply source/includes/stackpath/_workloads.md]: made the change in next commit 👍

**Review by y0geshdev — APPROVED** [review]
Thanks for the changes.

**Review by danielvanacker — APPROVED** [review]

**Pull request merged**